src/notebook.c

```diff
@@ -455,6 +455,30 @@ static void on_close_documents_right_activate(GtkMenuItem *menuitem, GeanyDocume
 }
 
 
+static void on_copy_filename_to_clipboard_activate(GtkMenuItem *menuitem, GeanyDocument *doc)
+{
+		g_return_if_fail(doc->is_valid);
+
+		gtk_clipboard_set_text(gtk_clipboard_get(GDK_NONE), g_path_get_basename(doc->real_path), -1);
```

Comment:
`g_path_get_basename()` returns an allocated string that needs to be `g_free()`'d.

Reply:
Im not GTK C or C fluent but this would entail declaring a separate `gchar var`, assigning the result of `g_path_get_basename()` to that var. Then `g_free(var)`, correct?

Reply:
Yep, except it needs to be a pointer (`*`) to the first character of the sequence of characters that makes up the string. This should probably do (untested):

```c
gchar *path = g_path_get_base(doc->real_path);
gtk_clipboard_set_text(gtk_clipboard_get(GDK_NONE), path, -1);
g_free(path);
```

Reply:
Gotta make an alteration. I also dont have my build server setup but ill set it up to test it